src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java

```diff
@@ -163,12 +167,30 @@ public String getInstanceUniqueId() {
         return String.valueOf(name.hashCode());
     }
 
+    private ComputeClient createClient() {
+        try {
+            ClientFactory clientFactory = new ClientFactory(Jenkins.get(), new ArrayList<>(),
+                    credentialsId);
+            return clientFactory.compute();
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Exception why creating GCE client", e);
+            return null;
+        }
+    }
+
     /**
      * Returns GCP client for that cloud.
      *
      * @return GCP client object.
      */
     public ComputeClient getClient() {
+        if (client == null) {
+            synchronized (this) {
+                if (client == null) {
+                    client = createClient();
```

Comment:
Since this is the only call site, suggest in-lining logic here.

Reply:
We should also handle the case when client creation failed either by throwing an exception or else somehow surfacing this in the UI.

Reply:
Its called in so many places that I dont even want to know what will happen if I will throw exception here..
Probably bad things 👎

Reply:
I suppose we would want to know though if there was an error retrieving the client. How can we best handle this?

Reply:
Previously there were nothing..
Now there is at least log.. :)

I would not like to tackle it in this PR..